gateways/adyen/adyen.go

```diff
@@ -75,6 +75,23 @@ func (client *AdyenClient) Authorize(request *sleet.AuthorizationRequest) (*slee
 			if cvcRaw, isPresent := values["cvcResultRaw"]; isPresent {
 				response.CvvResultRaw = cvcRaw.(string)
 			}
+			// if rtau status is present, expiry and last 4 must also be present
+			if rtauStatus, isPresent := values["realtimeAccountUpdaterStatus"].(string); isPresent {
+				rtauResponse := sleet.RTAUResponse{
+					RealTimeAccountUpdateStatus: GetRTAUStatus(rtauStatus),
+				}
+				if expiryDate, isPresent := values["expiryDate"].(string); isPresent {
```

Comment:
I see that expiry and last4 should be present if rtauStatus is present, but, since we check if the expiry and last4 values are present anyway, would the code look cleaner if we had three separate if-clauses for rtauStatus, expiry, and last 4? Or should we return an error if expiry and last4 are not present when rtauStatus is since that goes against our assumption?

Reply:
yeah that's what i was originally doing but wasn't sure if it was overkill -- I'll have 3 separate if conditions and then a check something like `rtauStatusOk && (!expiryDateOk || !lastFourOk)`

Reply:
What about checking that assumption in hail and logging an error if it's false? That way we don't return an error from sleet and think the transaction failed when in reality it may have been successful but we just got an abnormal rtau response

Reply:
okay sounds good -- in hail i can just check the `RTAUResponse` struct